### PR TITLE
[Sparkjob] Support deploying default image

### PR DESCRIPTION
**Pull request by benbd86**
added method deploy_default_image to sparkjob
also added a flag to use the default image explicitly

### CODE REVIEW

File: mlrun/runtimes/sparkjob.py [hunk 1]
```diff
@@ -170,20 +172,44 @@ class SparkRuntime(KubejobRuntime):
...
     kind = "spark"
     plural = "sparkapplications"
-
-    @property
-    def _default_image(self):
+    default_mlrun_image = ".spark-job-default-image"
+    gpu_suffix = "-cuda"
+
+    @classmethod
+    def _default_mlrun_image_name(cls, with_gpu=False):
```
Comment (urihoenig) [inline mlrun/runtimes/sparkjob.py:179]: I guess that as it's not a property it should be named _get...

File: mlrun/runtimes/sparkjob.py [hunk 2]
```diff
@@ -170,20 +172,44 @@ class SparkRuntime(KubejobRuntime):
...
+        suffix = cls.gpu_suffix if with_gpu else ""
+        return cls.default_mlrun_image + suffix
+
+    @classmethod
+    def deploy_default_image(cls, with_gpu=False):
+        from mlrun.run import new_function
+
+        sj = new_function(kind=cls.kind, name="spark-default-image-deploy-temp")
+        sj.spec.build.image = cls._default_mlrun_image_name(with_gpu)
+        if with_gpu:
```
Comment (urihoenig) [inline mlrun/runtimes/sparkjob.py:189]: I guess that you could shorten this like this, no?
sj.with_executor_requests(cpu=1, mem="512m", gpus=1 if with_gpu else 0)

File: mlrun/runtimes/sparkjob.py [hunk 3]
```diff
@@ -170,20 +172,44 @@ class SparkRuntime(KubejobRuntime):
...
-        gpu_bound = bool(driver_gpu or executor_gpu)
+        return bool(driver_gpu or executor_gpu)
+
+    @property
+    def _default_image(self):
         if config.spark_app_image_tag and config.spark_app_image:
             return (
                 config.spark_app_image
-                + ("-cuda" if gpu_bound else "")
+                + ("-cuda" if self._is_using_gpu() else "")
```
Comment (urihoenig) [inline mlrun/runtimes/sparkjob.py:212]: Why not use the gpu_suffix?

File: mlrun/runtimes/sparkjob.py (hunk 1 above)
Comment (urihoenig) [reply mlrun/runtimes/sparkjob.py:179]: In addition, I'm not sure the name is clear. Maybe default deployed image...

**Review by urihoenig — APPROVED** [review]

**Review by Hedingber — APPROVED** [review]

**Pull request merged**